plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -111,6 +114,57 @@ public function setTitle($title)
     }
 
     /**
+     * @param string $key
+     * @param string $value
+     * @param bool $override
+     *
+     * @return bool
+     */
+    public function addQueryParameter($key, $value, $override = true)
+    {
+        if (!isset($this->queryParameters[$key]) || true === $override) {
+            $this->queryParameters[$key] = $value;
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array $items
+     * @param bool|true $override
+     */
+    public function addQueryParameters(array $items, $override = true)
+    {
+        foreach ($items as $key => $value) {
+            $this->addQueryParameter($key, $value, $override);
+        }
+    }
+
+    /**
+     * @param $key
+     */
+    public function removeQueryParameter($key)
+    {
+        unset($this->queryParameters[$key]);
+    }
+
+    /**
+     * @param bool $urlEncode
+     * @return array
+     */
+    public function getQueryParameters($urlEncode = false)
+    {
+        if (!$urlEncode) {
+            return (array) $this->queryParameters;
+        }
+
+        return array_map(function($value) {
+            return !is_array($value) ? urlencode($value) : $value;
```

Comment:
I think a better method to use would be `Piwik\Common::sanitizeInputValues()` but I'm not 100% sure. @diosmosis is that correct that this method should be used maybe? Maybe even better would be to use `{{ queryParameters|e('url') }}` but according to [Twig docs](http://twig.sensiolabs.org/doc/filters/escape.html) it should not be used on entire URI but only a subcomponent being inserted. Not sure what they mean by subcomponent and entire URI.